### PR TITLE
Copy python tests for Modulemd.ModuleIndexMerger.merger into C

### PR DESCRIPTION
**Pull request by chenliTW**
For issue #198

### CODE REVIEW

File: modulemd/tests/test-modulemd-merger.c [hunk 1]
```diff
@@ -20,6 +21,7 @@
 #include "modulemd-module-index.h"
 #include "private/test-utils.h"
 
+typedef int gint;
```
Comment (OrionStar25) [inline modulemd/tests/test-modulemd-merger.c]: You could also use `guint64` and not use this at all. This is all defined in `glib` already.

Comment (OrionStar25) [inline modulemd/tests/test-modulemd-merger.c]: You could also use `guint64` and not use this at all. This is all defined in `glib` already.

File: modulemd/tests/test-modulemd-merger.c [hunk 2]
```diff
@@ -102,6 +104,164 @@ merger_test_deduplicate (CommonMmdTestFixture *fixture,
...
+
+  /*
+   * Test that override_index at a higher priority level succeeds
+   * Test that adding both of these at the same priority level fails
+   * with a merge conflict.
+   * Use randomly-selected high and low values to make sure we don't have
+   * sorting issues.
+   */
+  merger = modulemd_module_index_merger_new ();
+  random_low = rand()%100+1;
```
Comment (OrionStar25) [inline modulemd/tests/test-modulemd-merger.c]: Please look at this on how to use `random` with `GLib`. https://developer.gnome.org/glib/stable/glib-Random-Numbers.html

Declare your variable datatypes accordingly.

Comment (OrionStar25) [inline modulemd/tests/test-modulemd-merger.c]: Please look at this on how to use `random` with `GLib`. https://developer.gnome.org/glib/stable/glib-Random-Numbers.html

Declare your variable datatypes accordingly.

File: modulemd/tests/test-modulemd-merger.c [hunk 3]
```diff
@@ -12,6 +12,7 @@
  */
 
 #include <glib.h>
+#include <stdlib.h>
```
Comment (OrionStar25) [inline modulemd/tests/test-modulemd-merger.c]: I'm assuming you added this for `rand`. This can be eliminated by using the functionality in `glib`

**Comment by chenliTW** [comment]
I have run ```ninja test``` and change the ```rand()``` into ``g_random_int_range()```

**Comment by OrionStar25** [comment]
Please see the log from the failing CI. You have a bunch of warnings because of incompatible data types. Either typecast your parameters or use functions from the `glib` manual accordingly. We need to eliminate all the warnings. 

Also, you must run `ninja test` *right* before committing. Running `ninja test` after committing or before making another change would only fail the CI.

**Comment by chenliTW** [comment]
I have typecast all the parameters and eliminate all the warnings. But the CI test seems failing while I  run ```ninja test``` right before committing . Are there something that I missed to correct?

**Comment by chenliTW** [comment]
Th

> I have typecast all the parameters and eliminate all the warnings. But the CI test seems failing while I run `ninja test` right before committing . Are there something that I missed to correct?

The CI test passed after I use clang-format .

**Pull request merged**